src/systemd-sonic-generator/systemd-sonic-generator.c

```diff
@@ -104,7 +104,23 @@ static int get_target_lines(char* unit_file, char* target_lines[]) {
 static bool is_multi_instance_service(char *service_name){
     int i;
     for(i=0; i < num_multi_inst; i++){
-        if (strstr(service_name, multi_instance_services[i]) != NULL) {
+        /*
+         * The service name may contain @.service or .service. Remove these
+         * postfixes and extract service name. Compare service name for absolute
+         * match in multi_instance_services[].
+         * This is to prevent services like database-chassis and systemd-timesyncd marked
+         * as multi instance services as they contain strings 'database' and 'syncd' respectively
+         * which are multi instance services in multi_instance_services[].
+         */
+        char *saveptr;
+        char *token = strtok_r(service_name, "@", &saveptr);
+        if (token) {
+            if (strstr(token, ".service") != NULL) {
+                /* If we are here, service_name did not have '@' delimiter but contains '.service' */
```

Comment:
Do we need this if condition here? or is this added for additional check?

Ideally, we will not enter this condition as "service_name" that is being passed to this function should not include '.service.
https://github.com/sonic-net/sonic-buildimage/blob/master/src/systemd-sonic-generator/systemd-sonic-generator.c#L215
Test code :
#include<string.h>
#include<stdio.h>
int main(){
        char service_name[] = "swss.service";
        char *save_ptr2;
        char *type, *service;
        service = strtok_r(service_name, ".", &save_ptr2);
        type = strtok_r(NULL, "\n", &save_ptr2);
        printf("\nservice=%s, type=%s", service, type);
}
output:
service=swss, type=service
The value is service is passed to is_multi_instance_service()

Reply:
@anamehra  - this comment is pending, the other changes look ok to me.

Reply:
I saw <>.service being passed in some instances in my UT and that is why I added this additional check.
before L215, it is doing a strdup on word and using service_name to run strtok_r but passing word.

There is another instance https://github.com/sonic-net/sonic-buildimage/blob/master/src/systemd-sonic-generator/systemd-sonic-generator.c#L267

Reply:
Hi @SuvarnaMeenakshi , please let me know if you have any more queries. Thanks

Reply:
Got it, thank you for the pointer. 
Seems like this  change will handle both situation on service_name.type or service_name being passed.